extension/test/test-logger.js

```diff
@@ -10,7 +10,13 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
   var message = '';
 
   return function (status) {
-    var line = {}, hasMore;
+    var line = {}, hasMore, isSuccessStatus;
+
+    isSuccessStatus = typeof status == "number";
+    test.assert(isSuccessStatus, status["message"] ? status.message : status);
+    if (!isSuccessStatus) {
+      return;
+    }
```

Comment:
So the best way here would be to refactor the methods and let `_writeAsync()` return the promise, and not let it call a callback. That way you can better evaluate the result of the write operation, directly in the test and without all those workarounds.